### PR TITLE
Get a registry from GitHub (potentially with LocalRepos) on your REPL

### PR DESCRIPTION
**Pull request by nickjalbert**
This PR adds the ability to pull a Component from a registry file on GitHub that may or may not contain references to a LocalRepo.  For example, it allows the following demos on the REPL using what exists in the agentos GitHub.

Getting and running the `sb3_agent` on the REPL:

```python
from agentos.component import Component
component = Component.from_github_registry("https://github.com/agentos-project/agentos/blob/master/example_agents/sb3_agent/components.yaml", "sb3_agent")
component.run('evaluate')
```
Getting and running the `acme_r2d2` on the REPL (uglier because of ParameterSet):

```python
from agentos.component import Component
from agentos.parameter_set import ParameterSet
import yaml
import requests
component = Component.from_github_registry("https://github.com/agentos-project/agentos/blob/master/example_agents/acme_r2d2/components.yaml", "agent")
parameters = ParameterSet(yaml.safe_load(requests.get('https://github.com/agentos-project/agentos/raw/master/example_agents/acme_r2d2/parameters.yaml').content))
component.run('evaluate', parameters)
```

## Notes

* Not so bad in terms of actual code changes required to support.
* Had to swap out LocalRepos for GitHubRepos in the registry file.  ~~This is ugly and probably not a general operation.  Additionally the path relativization and versioning complicates the matter.  For the former, the proper move is probably to update all Components that used the swapped repo with the relativized path.  For the latter, maybe a new Repo type (VersionedGitHubRepo)?  It is a bit difficult to reason about the repos and registries when working at this level.~~
* Dealing with VirtualEnv is a bit annoying; the current solution is not great (activating and then losing the pointer).   Maybe Components should have a pointer to venv and `run()` activates/deactivates if the proper flag is passed?
* A lot of ~~fragile~~ somewhat-robust URL twiddling.

### CODE REVIEW

**Comment by nickjalbert** [comment]
@andyk Reworked this a bit.  The surface area of the change is pretty small and (I think) it enables brief and interesting demos (see PR description).  How would you feel about including this (or something like this)?

**Review by andyk — APPROVED** [review]
LGTM! Just nits, feel free to ignore. Merge away! I tried it locally on my MBP

File: agentos/utils.py [hunk 1]
```diff
@@ -5,7 +5,36 @@
 
 
 AOS_CACHE_DIR = Path.home() / ".agentos_cache"
```
[the remark this answers is not in the document]
Comment (nickjalbert) [reply agentos/utils.py:7]: Works for me!

File: agentos/utils.py [hunk 2]
```diff
@@ -5,7 +5,36 @@
...
+
+        https://github.com/<project>/<repo>/{blob,raw}/<branch>/<path>
+
+    and returns a 3-tuple of
+
+    1. The repo URL (i.e. the https URL the repo can be cloned from)
+    2. The branch_name or commit hash contained in the URL
+    3. The path of the file contained in the suffix of the URL
+    """
+    URL_STARTS_WITH = "https://github.com"
```
[the remark this answers is not in the document]
Comment (nickjalbert) [reply agentos/utils.py:24]: Sounds good.  I know there's another place we futz with GH urls in the codebase, maybe I'll see if we can combine logic.

**Comment by nickjalbert** [comment]
Comments addressed.  Merging!

Thanks for the review @andyk

**Pull request merged**